hooks/useThemeSettings/useThemeSettings.ts

```diff
@@ -0,0 +1,37 @@
+import { useNewsroomContext } from '@prezly/theme-kit-nextjs';
+import { useRouter } from 'next/router';
+import { useEffect } from 'react';
+import { useSessionStorage } from 'react-use';
+
+import type { ThemeSettings } from 'types';
+
+import { DEFAULT_THEME_SETTINGS, STORAGE_KEY } from './constants';
+import { parseQuery } from './utils';
+
+export function useThemeSettings(): ThemeSettings {
+    const { query } = useRouter();
+    const { themePreset } = useNewsroomContext();
+    const [previewSettings, setPreviewSettings] = useSessionStorage(STORAGE_KEY, {});
+
+    const settings: ThemeSettings = {
+        ...DEFAULT_THEME_SETTINGS,
+        ...themePreset?.settings,
+        ...previewSettings,
+    };
```

Comment:
How does `useMemo` work in case of dependencies being objects? I mean, if `previewSettings` is always a new object wouldn't that always trigger a call?

I have to check if that's the case.

Reply:
Turns out it's not possible to use `useMemo` because the object references change, mainly `query` object and possibly the `previewSettings` as well. Therefore the value is always recalculated.

Unfortunately this also means that the `useEffect` tries to parse the query object on every render but I've added an [early return](https://github.com/prezly/theme-nextjs-bea/pull/87/commits/3908436c72284a304160ff4633cf263ee7401144) to at least skip the parsing.

All in all, I don't think it's a big deal 🙂